src/components/AddTaskContainer.tsx

```diff
@@ -0,0 +1,61 @@
+import {
+  ChangeEvent,
+  Dispatch,
+  FormEvent,
+  SetStateAction,
+  useState,
+} from 'react';
+import { addTask, getTasks } from '../utils/api';
+import { Task } from '../utils/types';
+import AddTaskUI from './AddTaskUI';
+
+interface AddTaskContainerProps {
+  setTasks: Dispatch<SetStateAction<Task[]>>;
```

Comment:
🤔 Question: Do you really care that `setTasks` is a setState action? What if you wanted it to be some other updater function in the future?

Reply:
updated to use generic `setUpdateFunction`